src/open_inwoner/qmatic/client.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime
+
+from ape_pie.client import APIClient
+from pydantic import BaseModel
+from zgw_consumers.client import build_client
+from zgw_consumers.models import Service
+
+from open_inwoner.utils.api import JSONEncoderMixin
+
+from .exceptions import QmaticException
+from .models import QmaticConfig
+
+# API DATA DEFINITIONS
+
+
+class ServiceDict(BaseModel):
+    publicId: str
+    name: str
+    # could be float too in theory, documentation is not specific (it gives an int example)
+    duration: int
+    additionalCustomerDuration: int
+    custom: str | None
+
+
+class FullServiceDict(ServiceDict):
+    active: bool
+    publicEnabled: bool
+    created: int
+    updated: int
+
+
+class ServiceGroupDict(BaseModel):
+    services: list[ServiceDict]
+
+
+class BranchDict(BaseModel):
+    branchPublicId: str
+    branchName: str
+    serviceGroups: list[ServiceGroupDict]
+
+
+class BranchDetailDict(BaseModel):
+    name: str
+    publicId: str
+    phone: str | None
+    email: str | None
+    branchPrefix: str | None
+
+    addressLine1: str | None
+    addressLine2: str | None
+    addressZip: str | None
+    addressCity: str | None
+    addressState: str | None
+    addressCountry: str | None
+
+    latitude: float | None
+    longitude: float | None
+    timeZone: str
+    fullTimeZone: str
+    custom: str | None
+    created: int
+    updated: int
+
+
+class Appointment(JSONEncoderMixin, BaseModel):
+    services: list[ServiceDict]
+    title: str
+    start: datetime
+    end: datetime
+    created: int
+    updated: int
+    publicId: str
+    branch: BranchDetailDict
+    notes: str | None
+
+    class Config(JSONEncoderMixin.Config):
+        pass
+
+
+class NoServiceConfigured(RuntimeError):
+    pass
+
+
+# API CLIENT IMPLEMENTATIONS, per major version of the API
+
+
+def QmaticClient() -> "Client":
+    """
+    Create a Qmatic client instance from the database configuration.
+    """
+    config = QmaticConfig.get_solo()
+    assert isinstance(config, QmaticConfig)
+    if (service := config.service) is None:
+        raise NoServiceConfigured("No Qmatic service defined, aborting!")
+    assert isinstance(service, Service)
+    return build_client(service, client_factory=Client)
+
+
+def startswith_version(url: str) -> bool:
+    if url.startswith("v1/"):
+        return True
+    if url.startswith("v2/"):
+        return True
+    return False
```

Comment:
Instead of making this a boolean check, you could make it return or create an appropriate url, like:
```
def get_or_create_url_with_version(url):
    if url.startswith("v1/"):
        return url
    if url.startswith("v2/"):
        return url
    return f"v1/{url}"
```
Then below in `Client`: ```url = get_or_create_url_with_version(url)```

Reply:
Hmm I think I'll keep it as it is now, that way the function is a bit more clear in what it does imo